### PR TITLE
Entity parents.

### LINKED ISSUE #20: Parent Entity relationship
Allow an Entity to be related to a parent Entity (but limited to a single level, that is a parent cannot also be a child).

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**
Add the ability to make an Entity a parent of another Entity.
 - Closes #20.

### CODE REVIEW

File: app/views/entities/_form.html.erb [hunk 1]
```diff
@@ -7,3 +8,42 @@
...
                :boolean_style => :inline %>
+<%= form.input :parent_entity_id,
+               :as => :typeahead,
+               :wrapper_html => { :class => 'parent-entity' },
+               :label => 'Parent' %>
+
+<%= content_for :scripts do %>
+<% # This is horrible, how do we clean this up? %>
+<script>
+var entities = new Bloodhound({
```
Comment (michaelachrisco) [inline app/views/entities/_form.html.erb]: You can try https://github.com/LeaVerou/awesomplete

File: spec/features/user_changes_entity_parent_spec.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  context 'when we assign a parent' do
+    given(:parent_entity) { entities[1] }
+
+    scenario 'the parent is assigned' do
+      visit edit_entity_path(entity)
+
+      expect(page).to have_content('Edit Entity')
+
+      # REVIEW: Add javascript support so we can test the typeahead as well.
```
Comment (michaelachrisco) [inline spec/features/user_changes_entity_parent_spec.rb]: http://qunitjs.com/

Its easy to set up for ember. I would think that jQuery is just as easy.

File: spec/features/user_changes_entity_parent_spec.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  given(:entities) do
+    [
+      create(:entity),
+      create(:entity)
+    ]
+  end
+  given(:entity) { entities[0] }
+  given(:user) { create(:user, :all_roles) }
+
+  # TODO: Add test for using self as parent
```
Comment (michaelachrisco) [inline spec/features/user_changes_entity_parent_spec.rb]: Are you planning on adding these tests?

File: app/views/entities/_form.html.erb (hunk 1 above)
Comment (josephbridgwaterrowe) [inline app/views/entities/_form.html.erb]: I don't see support for remote data?

File: spec/features/user_changes_entity_parent_spec.rb (hunk 3 above)
Comment (josephbridgwaterrowe) [inline spec/features/user_changes_entity_parent_spec.rb]: I think I'm going to remove these remarks as these scenarios are covered by the model validation.

File: app/views/entities/_form.html.erb (hunk 1 above)
Comment (michaelachrisco) [inline app/views/entities/_form.html.erb]: You could put in a hook into javascript with a small api but that seems like overkill. 
Here is a similar issue: https://github.com/LeaVerou/awesomplete/issues/15

Comment (josephbridgwaterrowe) [inline app/views/entities/_form.html.erb]: Yes, I'm going to stick with typeahead for the time being... I'm open to looking at others that support remote data sources.

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco did I address all your feedback or did I miss something?

**Comment by michaelachrisco** [comment]
Were you able to test the entities form typeahead? That is the only thing I am waiting on at this point.

**Comment by josephbridgwaterrowe** [comment]
No, and I'm not going to. :smile: 

It was painful and sucking my will and wasn't really worth the :hankey:

**Comment by michaelachrisco** [comment]
Aww and here I like javascipt testing. Ill merge it for now but I might take a stab at it sometime for you :P

**Pull request merged**